pkg/syncer/volume_health.go

```diff
@@ -77,58 +77,75 @@ func csiGetVolumeHealthStatus(ctx context.Context, k8sclient clientset.Interface
+				newPvc, err := k8sclient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(ctx, pvc.Name, metav1.GetOptions{})
+				if err == nil {
+					log.Infof("updateVolumeHealthStatus: updating volume health annotation for pvc %s/%s which "+
+						"get from API server from old value %s to new value %s at time %s",
+						newPvc.Namespace, newPvc.Name, val, volHealthStatus, time.Now().Format(time.UnixDate))
+					metav1.SetMetaDataAnnotation(&newPvc.ObjectMeta, annVolumeHealth, volHealthStatus)
+					metav1.SetMetaDataAnnotation(&newPvc.ObjectMeta, annVolumeHealthTS, time.Now().Format(time.UnixDate))
+					_, err := k8sclient.CoreV1().PersistentVolumeClaims(newPvc.Namespace).Update(ctx, newPvc, metav1.UpdateOptions{})
 					if err != nil {
-						if apierrors.IsConflict(err) {
-							log.Debugf("csiGetVolumeHealthStatus: Failed to update pvc %s/%s with err:%+v, will retry the update",
-								pvc.Namespace, pvc.Name, err)
-							// pvc get from pvcLister may be stale, try to get updated pvc which bound to pv from API server
-							newPvc, err := k8sclient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(ctx, pvc.Name, metav1.GetOptions{})
-							if err == nil {
-								log.Infof("csiGetVolumeHealthStatus: updating volume health annotation for pvc %s/%s which "+
-									"get from API server from old value %s to new value %s at time %s",
-									newPvc.Namespace, newPvc.Name, val, volHealthStatus, time.Now().Format(time.UnixDate))
-								metav1.SetMetaDataAnnotation(&newPvc.ObjectMeta, annVolumeHealth, volHealthStatus)
-								metav1.SetMetaDataAnnotation(&newPvc.ObjectMeta, annVolumeHealthTS, time.Now().Format(time.UnixDate))
-								_, err := k8sclient.CoreV1().PersistentVolumeClaims(newPvc.Namespace).Update(ctx, newPvc, metav1.UpdateOptions{})
-								if err != nil {
-									log.Errorf("csiGetVolumeHealthStatus: Failed to update pvc %s/%s with err:%+v",
-										newPvc.Namespace, newPvc.Name, err)
-								}
-							} else {
-								log.Errorf("csiGetVolumeHealthStatus: volume health annotation for pvc %s/%s is not updated because "+
-									"failed to get pvc from API server. err=%+v",
-									pvc.Namespace, pvc.Name, err)
-							}
-						} else {
-							log.Errorf("csiGetVolumeHealthStatus: Failed to update pvc %s/%s with err:%+v",
-								pvc.Namespace, pvc.Name, err)
-						}
+						log.Errorf("updateVolumeHealthStatus: Failed to update pvc %s/%s with err:%+v",
+							newPvc.Namespace, newPvc.Name, err)
 					}
+				} else {
+					log.Errorf("updateVolumeHealthStatus: volume health annotation for pvc %s/%s is not updated because "+
+						"failed to get pvc from API server. err=%+v",
+						pvc.Namespace, pvc.Name, err)
 				}
+			} else {
+				log.Errorf("updateVolumeHealthStatus: Failed to update pvc %s/%s with err:%+v",
+					pvc.Namespace, pvc.Name, err)
 			}
 		}
 	}
-	log.Infof("GetVolumeHealthStatus: end")
+
+	log.Infof("updateVolumeHealthStatus: end")
```

Comment:
I would recommend to remove this log. And moreover this is info log and it could potentially fill up the logs at customer env.

Reply:
Done